src/gkeClient.ts

```diff
@@ -270,16 +277,17 @@ export class ClusterClient {
         `Failed to get project ID for cluster membership discovery. Try setting "project_id".`,
       );
     }
+
     const url = `${this.hubEndpoint}/projects/${projectID}/locations/global/memberships?filter=endpoint.gkeCluster.resourceLink="${clusterResourceLink}"`;
     const resp = (await this.auth.request({
       url: url,
       headers: {
         'User-Agent': userAgent,
       },
-    })) as HubMembershipsListResponse;
+    })) as HubMembershipsResponse;
 
     const memberships = resp.data.resources;
-    if (memberships.length < 1) {
+    if (!memberships || memberships.length < 1) {
```

Comment:
This was a bug I found in testing. Sometimes the response comes back as null instead of `[]`, which then NPEd on `.length`.